backend/src/test/java/de/neuefische/backend/controller/RecipeControllerTest.java

```diff
@@ -147,4 +148,67 @@ void getAllRecipe_returnsRecipeWithAllChoicesAvailable_whenCalledWithRecipeInRep
                 .andReturn();
     }
 
+    @Test
+    void getRecipeById_returnsRecipeWithId1_whenCalledWithId1() throws Exception {
+        //GIVEN
+        PreparationTime preparationTime = new PreparationTime(0, 30);
+        TotalTime totalTime = new TotalTime(1, 15);
+        List<RecipeIngredients> recipeIngredients = new ArrayList<>();
+        recipeIngredients.add(new RecipeIngredients("name test", "quantity 1"));
+        recipeIngredients.add(new RecipeIngredients("name test 2", "quantity 2"));
+        Recipe recipe = new Recipe("1",
+                "Test Recipe",
+                "Test Description",
+                "Test Instructions",
+                "Test Author",
+                "Test Origin",
+                List.of(RecipeType.VEGETARIAN, RecipeType.WITH_MEAT),
+                preparationTime,
+                totalTime,
+                List.of(RecipeCategory.DINNER, RecipeCategory.SIDE_DISH),
+                RecipeDifficulty.EASY,
+                recipeIngredients
+        );
+        repo.save(recipe);
+        //WHEN & THEN
+        mvc.perform(get("/api/recipes/1"))
+                .andExpect(status().isOk())
+                .andExpect(content().json("""
```

Comment:
You will make your life a lot easier, if you use an ObjectMapper:

After
```
    @Autowired
    private MockMvc mvc;
```
add
```
    @Autowired
    private ObjectMapper objectMapper;
```

Then in the test:
`String recipeJson = objectMapper.writeValueAsString(recipe)`

and
`content(recipeJson)` instead of `content().json(...)`

You can also decode a response with an ObjectMapper:
```MvcResult resultJson = mvc.perform(...).andReturn();

Recipe returnedRecipe = objectMapper.readValue(resultJson.getResponse().getContentAsString(), Recipe.class);```